com.unity.ml-agents/Runtime/Communicator/GrpcExtensions.cs

```diff
@@ -237,19 +237,32 @@ public static ObservationProto GetObservationProto(this ISensor sensor, Observat
             var shape = sensor.GetObservationShape();
             ObservationProto observationProto = null;
             var compressionType = sensor.GetCompressionType();
-            // Check capabilities if we need to concatenate PNGs
-            if (compressionType == SensorCompressionType.PNG && shape.Length == 3 && shape[2] > 3)
+            if (!s_HaveWarnedAboutTrainerCapabilities)
```

Comment:
This is incorrect - previously it would "downgrade" to uncompressed observations every time; now it does it the first time but not subsequent times, so training will break.

Reply:
Probably best to rename this and add a new flag for the compression mapping warning.